### ISSUE TITLE
Training from scratch with dice loss only

### ISSUE THREAD

**Issue by fiskrt** [hkchengrex/XMem#62]
Hello, I'm trying to train the model with only dice loss (i.e no BCE loss) but the dice loss does not improve as you can see in the figure below. Pink is dice only, blue is dice+BCE.

![image](https://user-images.githubusercontent.com/43207511/216840955-93abb2cf-7dfd-43f9-945f-fa4097352544.png)

These are the resulting masks, where the dice-only-masks (left) are completely uniform and does not improve at all.
![image](https://user-images.githubusercontent.com/43207511/216840601-5580a00e-9b11-46f3-923c-7aabe48820e1.png)

To remove BCE I just commented line [59-64](https://github.com/hkchengrex/XMem/blob/139a7458b84a4829cb3bc7434cea1e7383eca5f8/model/losses.py#L58#L64) in `losses.py`. I'm using a batch size of 4 and training the model from scratch (no pre-trained checkpoint) using
`
python -m torch.distributed.run --master_port 25763 --nproc_per_node=1 train.py --exp_id dav_yt_only_dice --stage 3
`

Do you have any suggestions on what I might be doing wrong? Thanks

**Comment by hkchengrex** [comment]
Not sure why would you want to do that... From experience, it can converge with BCE loss alone. I have not tried not using BCE.

**Comment by fiskrt** [comment]
I'm researching another type of loss function which is similar to the dice loss.
So I want to make sure I can train the network with only the dice loss at first. So even if the accuracy is not that good using only dice, it would be really nice to get some mask predictions that are not completely uniform.

Do you have any ideas on what I could try to get some non-uniform mask predictions?

**Comment by hkchengrex** [comment]
I would start by changing the learning rate.
